### ISSUE TITLE
New logo?

### ISSUE THREAD

**Issue by aidanholm** [luakit/luakit#363]
With the new 2017 release, it'd be nice to have a fresh logo. Preferably something with moon imagery.

**Comment by aidanholm** [comment]
Flat-ish design:

1a) ![design_1_blue](https://user-images.githubusercontent.com/10691965/29616473-3c58df76-8844-11e7-9d49-ff40024e4c6c.png) 1b) ![design_1_purple](https://user-images.githubusercontent.com/10691965/29616474-3e0775f8-8844-11e7-847c-778a6796b726.png) 1c) ![design_1_dark](https://user-images.githubusercontent.com/10691965/29620766-70ee82a6-8851-11e7-884a-2851358b7ab8.png)


Slightly more 3d design:

2a) ![design_2_black](https://user-images.githubusercontent.com/10691965/29632840-d369184a-8876-11e7-9f8a-bedf3d3b3c7c.png)2b) ![design_2_blue](https://user-images.githubusercontent.com/10691965/29616809-355cc39e-8845-11e7-9c4a-e10adea896b1.png)2c) ![design_2_purple](https://user-images.githubusercontent.com/10691965/29616816-39db741a-8845-11e7-98ef-b55ad59338ba.png)

**Comment by Plaque-fcc** [comment]
Nice but a bit faceless, I think. Can this Luna be, like, purple?

On August 23, 2017 1:51:01 PM GMT+03:00, Aidan Holm <notifications@github.com> wrote:
>Rough draft; I am clearly no designer, but it's a start:
>![icon](https://user-images.githubusercontent.com/10691965/29612410-c1f27112-8833-11e7-9a4b-e459339e993a.png)
>
>-- 
>You are receiving this because you are subscribed to this thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/luakit/luakit/issues/363#issuecomment-324293276

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by aidanholm** [comment]
I chose blue because it matches the website, but I'll play around with it; feel free to post modifed versions too
Edit: added a purple variation too

**Comment by aidanholm** [comment]
I think part of the problem with these designs might be that the crescent moon is too thin; need to play around with it more

**Comment by sionleroux** [comment]
What if the moon was part a sphere mesh usually used to represent the web, or some other kind of web-related imagery, for example Firefox has a fox around the world, and the old internet explorer icon had a blue letter e in a globe. So maybe replacing the earth in these ideas with the moon could be an idea. Like a web sphere, with a crescent moon. Not sure if that makes the icon too busy though, but it's just an idea to make it represent _"moon + web"_ and not just _"moon"_.

**Comment by aidanholm** [comment]
@sinisterstuf That's a good idea; I did some playing around with it and came up with these:

3 ) ![globe](https://user-images.githubusercontent.com/10691965/29748745-2d1393ea-8b4f-11e7-930e-e05de4848c96.png)

4 ) ![globe2](https://user-images.githubusercontent.com/10691965/29748792-0b407eb2-8b50-11e7-92d2-b619c752ac10.png)

I'm not sure the first one is as good as the other ones though. It may be that the moon covers too much of the map detail. The hybrid approach isn't too bad. The colours need some work for both. I'm not a designer, which is why I stuck to minimalist designs before :)

**Comment by Plaque-fcc** [comment]
Surely, this time I like the second one much more.

On August 27, 2017 12:52:21 PM GMT+03:00, Aidan Holm <notifications@github.com> wrote:
>@sinisterstuf That's a good idea; I did some playing around with it and
>came up with these:
>
>![globe](https://user-images.githubusercontent.com/10691965/29748745-2d1393ea-8b4f-11e7-930e-e05de4848c96.png)
>
>![globe2](https://user-images.githubusercontent.com/10691965/29748792-0b407eb2-8b50-11e7-92d2-b619c752ac10.png)
>
>I'm not sure the first one is as good as the other ones though. It may
>be that the moon covers too much of the map detail. The hybrid approach
>isn't too bad. The colours need some work for both. I'm not a designer,
>which is why I stuck to minimalist designs before :)
>
>-- 
>You are receiving this because you commented.
>Reply to this email directly or view it on GitHub:
>https://github.com/luakit/luakit/issues/363#issuecomment-325188364

-- 
J.K.

**Comment by aidanholm** [comment]
I think the second hybrid design is better as well. It's also a more portable concept: a crescent moon superimposed over a world map is easy to do in the style of any icon theme, and it's still easily identifiable. So it'd be easy to do a high-contrast version, a square version for a square icon theme, etc.

**Comment by hippi777** [comment]
hi there!

i think the original lua logo could be a better indicator for something with lua (so if someone can see only the logo, they will know it's a lua stuff)

my idea is to take right that earth image for a layer in the image editor, draw around the ground with straight lines connected with small dots, and where are more space, there can be other dots inside, so this makes an earth-like graph, and finally by removing the layer with the earth, the graph remains, and then just cut off a hole and put that in the right corner...

extra shine could come from making the graph extended to the oceans, but with different color, and the dashed line around the main circle in the lua logo could join as well into the graph with an another color in space and some insignificant color from them to the other parts of the graph can be added if that wont make a mess... maybe satellite like nodes there can pimp it up, but that will have disharmony... maybe the dashed line can be left out as well for a bigger main part of the thumbnail...

the lua logo can be used also as a layer to draw the dashed circle and cut off the moon part and place it where it is intended to be... this could have lua, earth and network all together, if it's not too messy then it can fit on a small pic as well, but that can have a smaller graph...

for colors, lua logo could be a good base, but needs more of them, probably the blue of the manual could be a good one, or could be harmonized...

(ps: sorry for still didn't have time to continue replacing jq, i have no time to breathe :D )

**Comment by aidanholm** [comment]
Hey :)

Sounds interesting, although it's difficult to tell what it would look like just from a textual description. Mockups welcome :) I'm concerned that it could lead to a design with lots of visual elements, which is something I'm trying to avoid. Ideally the new logo will be distinctive at all sizes, even at 16x16 (website favicon), and too much detail can make that difficult, even if it looks good at larger sizes. I tried some mockups that were based on the shape of the Lua logo, and none of my attempts were very successful.

I've been thinking, and I don't think using the shape of the Lua logo is a good idea. Most projects I've seen with that type of logo are Lua libraries, and luakit is a web browser, not a library. More importantly, luakit is more "web browser" than "lua program"; a big part of my efforts have been to make luakit easy to use without having to program it. For this reason, while I'd like some reference to Lua (hence the moon), I don't want an association with "writing Lua code"; I want luakit to have its own independent identity.

Re replacing jQuery: No problem, take your time :)

**Comment by hippi777** [comment]
i never had this consideration, about app vs lib but i think it could be meaningful, and even the kit name is like a lib of a toolkit, but knowing it's a browser, probably this is not in the head of the ppl... but probably could look good and not the 16x16px is the most important, that's enough to make it something like... it could be even more like a cool brand logo, but no offense for your idea, more likely not because i didn't come up with anything visual, cuz lack of time...

btw there souldn't be much nodes in the graph, but still dunno if compared to that earth which one would lose more details, i think the main circle is the most important, and that can reach the ends of the left and bottom frames in 16x16 and then there can be a small piece of the smaller circle in the corner like how kids drawing the sun :D 

but everything is upon you, i've just found it a good idea, i've already thought about it, what could look good, and now just came an idea...

**Comment by aidanholm** [comment]
You're right about the name "luakit"; it does have that feel to it. Luakit is a well-known name though, so I wouldn't want to change it and lose that recognition. I rather like the name myself :)

I think a globe is something that would work well as the backdrop of the icon; it's a very easily recognizable shape/pattern, and it's easy to drop detail at lower resolutions and turn it into a blue circle; still instantly recognizable. A graph is a lot more detailed, and it's also a less common concept in icons, which means it's not instantly recognizable. There's also a very common association with world globes and web browsers in icons, but a graph is ambiguous. For the same sort of reason, I'd also prefer a simple cresent moon rather than a Lua-logo-style circle moon: it's instantly recognizable as a moon shape.

So currently I'm favoring the second design here https://github.com/luakit/luakit/issues/363#issuecomment-325188364, because I think it fits all these design goals best. But more feedback and ideas are always welcome :)

**Comment by hippi777** [comment]
fine i've let go my idea (rip), btw i wasn't about arguing on the name, thats fine :)

**Comment by aidanholm** [comment]
You could always post a draft version :) yep, I know, no worries

**Comment by ghost** [comment]
I started to draw myself, but I'm so bad at it. I actually had that idea of a moon, which looks like gear on the outside. Maybe with a "technical drawing" overlay, like helper lines. This could symbolize it's customizability.

**Comment by sionleroux** [comment]
Nice idea @xkpd3, I think if I understand you correctly, the shape is a crescent moon with gear teeth on the outside, like this:

5 ) ![moongears](https://user-images.githubusercontent.com/840466/29761673-b120170a-8bca-11e7-84e4-d9c9be0e8df0.png)

I just hacked this together quickly by putting squares roughly on the edge of a circle shape, because a sketch is easier to think about, to see if this is what you mean, but this is obviously a very rough drawing. If this is the right direction, then maybe we could flesh it out with more detail and nicer shapes to make it look good.

Apart from that agree that so far the best is idea no. 2. (Maybe we should number these ideas).

[@aidanholm edit: added design number]

**Comment by aidanholm** [comment]
Good idea, I've added numbers

**Comment by Plaque-fcc** [comment]
Liked both what I mentioned before and №2.

On August 28, 2017 9:31:31 AM GMT+03:00, "Siôn Le Roux" <notifications@github.com> wrote:
>Nice idea @xkpd3, I think if I understand you correctly, the shape is a
>crescent moon with gear teeth on the outside, like this:
>![moongears](https://user-images.githubusercontent.com/840466/29761673-b120170a-8bca-11e7-84e4-d9c9be0e8df0.png)
>I just hacked this together quickly by putting squares roughly on the
>edge of a circle shape, because a sketch is easier to think about, to
>see if this is what you mean, but this is obviously a very rough
>drawing. If this is the right direction, then maybe we could flesh it
>out with more detail and nicer shapes to make it look good.
>
>Apart from that agree that so far the best is idea no. 2. (Maybe we
>should number these ideas).
>
>-- 
>You are receiving this because you commented.
>Reply to this email directly or view it on GitHub:
>https://github.com/luakit/luakit/issues/363#issuecomment-325271107

-- 
J.K.

**Comment by aidanholm** [comment]
Unless anyone posts something absolutely amazing (please do), I think I'll go with number 4. That's this one:

![globe2](https://user-images.githubusercontent.com/10691965/29748792-0b407eb2-8b50-11e7-92d2-b619c752ac10.png)

From here, I'll be mostly working to refine and tweak this design

**Comment by sionleroux** [comment]
I also agree that this is the best so far. I've asked a designer friend of mine to to take a look, maybe they'll have a nice idea...

**Comment by groolot** [comment]
Can you push the design 4 source file?

**Comment by aidanholm** [comment]
@groolot Uploaded as a text file because GitHub doesn't allow svg for some reason... [icon.svg.txt](https://github.com/luakit/luakit/files/1328967/icon.svg.txt)

**Comment by lostinlight** [comment]
Hi,
I came across this issue on Mastodon, and, though not a designer, decided to experiment a little.
Some sketches that came to mind:
6) ![logo ideas](https://gitlab.com/lostinlight/per_aspera_ad_astra/raw/master/luakit-concept/Luakit_ideas.jpg)

Colours are chosen randomly.
Uploaded png images and svg files [here](https://gitlab.com/lostinlight/per_aspera_ad_astra/tree/master/luakit-concept), in case someone wants to experiment further.

Also tried to imagine how the icons would look on a desktop ([image](https://gitlab.com/lostinlight/per_aspera_ad_astra/raw/master/luakit-concept/Luakit_demo.jpg)). Number 4 variant from the discussion above fits in. From the very first message 1c and 2c look nice.

**Comment by aidanholm** [comment]
These look good! I'll probably stick with the same design as before, but these mockups show how the same concept can be made to work in a multitude of styles

**Comment by lostinlight** [comment]
Great! If you ever need a design for an open source project, there's a wonderful initiative (perhaps you know about it) - [Open Source Design](http://opensourcedesign.net/jobs), where one can post a task and get pro bono help from the community. :)

**Comment by sionleroux** [comment]
I really like these @lostinlight, probably my favourite style so far.

**Comment by ghost** [comment]
Wow, I really like the dark version of c) @lostinlight 

Also  - awesome that you saw my mastodon toot!!! :+1:

**Comment by sionleroux** [comment]
Agreed, I'd definitely go for 6.b) or 6.c)
Although in the case of 6.b) maybe it would be even better without the grey outline on the moon?

**Comment by aidanholm** [comment]
I've been playing around with 6b and 6c, and unfortunately didn't have much success in developing them from concept to usable icon :( I really liked the colours of 6b, so I used those to modify no 4, making it flatter at the same time:

![icon-lg](https://user-images.githubusercontent.com/10691965/32133123-19292bd4-bc04-11e7-9741-7153f74dcad3.png)

How it looks on the website:

![out](https://user-images.githubusercontent.com/10691965/32133148-b9d5f238-bc04-11e7-9d34-7fb7e4ef1007.png)

**Comment by hippi777** [comment]
heyy it became nice! :D how it looks like in 16x16 and 32x32? even if this question is not something that can be a reason to throw away anything, i rly like this one, and its hard to reach that point with my taste, so this is a big plus :D

**Comment by hippi777** [comment]
however it could be LuaKit instead of luakit, thats some selfappreciate for a great product that i still didnt make default

offtopic

cuz i have a new job without running like before, but i wanna make my installer for all my stuffz together included that will handle setups like installs of different lua variants different versions different apps and different compiler options; and mostly for update, thats a must for luakit... currently i have big dirt around my project, so its a primer stuff to leave this behind, but its a hacky stuff cuz its in bash, which tells kinda much everything about hacky with its toolset (check sudo check installations check visiblility handling a flat tree of arguments, passing arrays as a single parameter to a function and whatever :D ) good news is that ive just killed firefox 1-2 weeks ago and ive lost ~200 tabs of research matery, so moving will be this much easier :D life is hard but im on the way to luakit

**Comment by aidanholm** [comment]
16x16 and 32x32 are both fairly distinctive, but they'll most likely need separate source svg files. A 16 px icon is probably too small to have a drop shadow, for example: it'd take up pixels best used to make the world/moon clearer.

Re name style: I've been thinking about this for a while, and I'm not terribly fond of 'luakit', but I especially dislike 'LuaKit'; it just doesn't seem balanced to me. So we will probably switch to 'Luakit', unless anyone really doesn't like that.

Your project sounds pretty challenging! Have you considered writing it in lua? :) You don't get the same direct interface to files and processes, but instead you get much better control structures.

**Comment by hippi777** [comment]
Luakit is already better, but LuaKit is nearer to programmers as it have a camelcase :D  - thats all i can say for that...

thats just my installer, its already working, just i wanna make it flexible and a complete tool for manage setups... current problems are to be sure if ive checked out sudo timeout it wont timeout when its used - just for change a text to display :D i cant decide if the rights with sudo are suitable for install sysdeps as sudores have very complex syntax, so i can disable sudo and use su -c instead. i'm not sure about the right method for every case about visibility, i can use -z PS1; ps $$ including '-'; tset -q; or $DISPLAY, and i wanna be stable with ssh, tty, terminal emulators and running without any display... i wanna make it able to use a conf file and bash params, conf file is straightforward, but for bash params i dunno if i should go through the whole list more than once or juggle with numbers in a numeric for or while loop... everything else is straightforward :D the flattened tree of arguments is like whatever: sub1 sub2 - whatelse - this can be easily followed balanced (the open+close for any levels) and extracted, its purpose is to make things like "LuaJIT: make: PREFIX=whatever - v=2.1 reinstall - TekUI" and then i can really make any setup without the need for LuaJIT_version=2.1, but i wanna make that possible, the config would be just like this, and that can be reused to extend modify and clone a project... dependencies must be detected so if i write TekUI then it should select LuaJIT as well for default if there is no lua luajit ravi openrestyluajit raptorjit or whatever selected... so this one is big, and i dont wanna go with luarocks or sys packages cuz the outdated versions all around... plus lazy forms like LuaJIT can be written as luajit as well mwhahahaaa :D

the reason for using bash, is that i wanna make it run instantly via a doubleclick (only after extract a tarball that keeps the executable flag - but even my username and group are kept... :/ ) i could have other scripting languages (no blobs allowed for shipping) but things like perl or python are not as sympathetic, and bash is something i can rely on on a fresh debian (i think it would fail and cut my path to go if dash would make bash not preinstalled cuz i have much bashisms) not even a 'make' is installed, but thats the same pain as bash... currently it uses the sudo/su for installing system packages related to a fresh debian (now only those which r missing, so wont say already installed and selected it as manually installed...) those r like make git hq dev headers and dunno... downloading lua related sources compiling and installing them under the project dir so no sudo require, but i'm thinking about syswide install, but this works even on a server without privileges, and on the same architecture i can freely carry the installed stuff in a single directory even put on a fresh debian and it just works, but then there are the binary blobs... making small checks and and running my project which have a small supervisor bash for getting the current directory and for making able to restart from within the app or run different stuffs like test mode for tekui (for getting errors if i wanna switch to a modified version) and here will exist the other aims like turbo server and tekui...

the whole stuff is a big apparat of tools like a swiss army knife, and then i can even combine them on variuos ways like share tools or handle everything from within tekui and whatever :D 

still takes much work even for these, and i still have 4k+ lines of plans.... one day it will be public, but i cant promise nothing, but i wanna get ppl involved before publication, but it have to get much strength before share to keep the directions of development and before public to not let others to hit and run and take it on other platforms poizion it with money team marketing and not good aims... however i think ill go with gpl and i think im able to give back to the mit and apache origins under those licences if ill make changes on those and have a commercial license to give back some money if they are making income with it, but im still not enough familiar with laws to be sure about these...

whenever this big stuff will be done extending it wont hurt, but it will give me back much time with automatized updates without killing existing working setups and much easier testing, i can even make then a default x64 and x32 setup to make it extra portable :D 

and i still dont care about win and mac :D farer is better, but i think both could use as win have linux inside and mac is posix... sadly, cuz i dont wanna reach the point when its ported to those and i wont touch those systems and therefore a stronger fork will take away my role wreck the directions of development and make me powerless and i have to figt alone even after ill wish for contributors... so its really a hard case... no win or no python would be good for my remaining plans...

the 1st public appearance will happen before private shares, as i wanna make a server based on it sooner or later, and for share it with others i need to make it able to communicate with other instances (i think ljcurl will do what i need and ddns and letsencrypt gonna be there before i can start the server) and as well i need to make some comfort for the coworking...

sooooo muuuuch to do.... :D at least not impossible...

**Comment by hippi777** [comment]
and sorry for the big off! :D

**Comment by groolot** [comment]
This logo will be use inside Debian package.

![luakit](https://user-images.githubusercontent.com/632571/34081162-833af1bc-e349-11e7-95fc-dc464729f5f4.png)
[luakit.zip](https://github.com/luakit/luakit/files/1565805/luakit.zip)

**Comment by aidanholm** [comment]
@groolot that looks like an older variant: the up-to-date one is https://github.com/luakit/luakit/issues/363#issuecomment-340164705

Also I haven't uploaded the SVG yet: I remember there are some minor touchups I need to do first

**Comment by groolot** [comment]
Is it possible to include the SVG source inside the repo to keep track of modifications and stop using the issue discussions to show it as the newer version :)

**Comment by McSinyx** [comment]
I totally agree: that way the logo will be licensed more publicly.

**Comment by aidanholm** [comment]
Pushed! feel free to open PRs if you want to do touchups; in the meantime, we can finally close this :)

**Comment by aidanholm** [comment]
@mason-larobina any chance we could get this set as the repo logo?
